.github/workflows/deploy.yml

```diff
@@ -0,0 +1,167 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+    - main
+
+jobs:
+  deploy_build:
+    name: deploy Build
+    runs-on: windows-latest
+    env:
+      solution_folder: solutions
+      solution_name: DQT20
+      solution_exported_folder: out/exported/
+    environment: Build
+    steps:
+    - name: gitConfig
+      run: |
+        git config --global core.longpaths true
+
+    - uses: actions/checkout@v2
+
+    - name: Pack unmanaged solution
+      uses: microsoft/powerplatform-actions/pack-solution@v0
+      with:
+        solution-folder: ${{ env.solution_folder }}/${{ env.solution_name }}
+        solution-file: ${{ env.solution_name }}.zip
+
+    - uses: Azure/login@v1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+    - uses: DfE-Digital/keyvault-yaml-secret@v1.0.1
+      id: CRM
+      with:
+        keyvault: ${{ secrets.KEYVAULT_NAME }}
+        secret: CRM
+        key: CRM_URL, CRM_APP_ID, CRM_APP_SECRET, CRM_TENANT_ID
+
+    - name: Import unmanaged and packed solution
+      uses: microsoft/powerplatform-actions/import-solution@v0
+      with:
+        environment-url: ${{ steps.CRM.outputs.CRM_URL }}
+        app-id: ${{ steps.CRM.outputs.CRM_APP_ID  }}
+        client-secret: ${{ steps.CRM.outputs.CRM_APP_SECRET  }}
+        tenant-id: ${{ steps.CRM.outputs.CRM_TENANT_ID  }}
+        solution-file: ${{ env.solution_name }}.zip
+
+    - name: Export managed solution
+      uses: microsoft/powerplatform-actions/export-solution@v0
+      with:
+        environment-url: ${{ steps.CRM.outputs.CRM_URL }}
+        app-id: ${{ steps.CRM.outputs.CRM_APP_ID  }}
+        client-secret: ${{ steps.CRM.outputs.CRM_APP_SECRET  }}
+        tenant-id: ${{ steps.CRM.outputs.CRM_TENANT_ID  }}
+        solution-name: ${{ env.solution_name }}
+        solution-output-file: ${{ env.solution_exported_folder }}/${{ env.solution_name }}_managed.zip
+        managed: true
+
+    - name: Publish crm artifact
+      uses: actions/upload-artifact@v2
+      with:
+        name: drop-crm
+        path: ${{ env.solution_exported_folder}}/${{ env.solution_name }}_managed.zip
+
+  import_test:
```

Comment:
The jobs look very similar. Could we use a matrix? https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix

Reply:
Changed the jobs to use matrix